tests/java/ast_java.re

```diff
@@ -564,3 +564,314 @@ and inferred_parameters = (string, intermediate7, string)
 and enum_body_declarations = (string, intermediate93)
 and synchronized_statement = (string, parenthesized_expression, block)
 and program = list(_statement);
+type cst_node = 
```

Comment:
@aryx i feel that there is too many duplicates caused by the fact that I create new type for each "CHOICE" i find in the grammar. This is hitting 
```
Too many non-constant constructors
     -- maximum is 246 non-constant constructors
```
I believe I have to have hierarchical type constructor definition here. Then the simple concat all the rule strings with "and" does not work for generating the types and their constructor!

Reply:
There is no need for this cst_node, so this problem should go away.